src/db_state.rs

```diff
@@ -171,7 +173,7 @@ impl SortedRun {
                 None
             };
 
-            if current_sst.intersects_range(upper_bound_key, range) {
+            if current_sst.intersects_range(upper_bound_key, range.clone()) {
```

Comment:
IIUC, `range.clone()` will result in a copy of the key range for every SR in every get call that might contain the key. That seems kinda expensive. Is there a way to play a trick with the lifetime so we don't need to clone() for intersects_range?  Or perhaps using a reference instead of ownership?

Reply:
Hmm, I think this is a shallow clone (just a copy of the references).

Reply:
Oh, TIL. I guess it's ref counting the underlying data on the heap?

Reply:
Deep.. nm. It's a slice so it's not owned 🤦‍♂️ been coding python too much this week